File: Solutions/1346-CheckIfNAndItsDoubleExist/CheckIfNAndItsDoubleExist.kt
class Solution {
    fun checkIfExist(arr: IntArray): Boolean {
        val set: MutableSet<Int> = mutableSetOf()

        for (num in arr) {
            if (set.contains(num * 2) || (num % 2 == 0 && set.contains(num / 2))) {
                return true
            } else {
                set.add(num)
            }
        }

        return false
    }
}

File: Solutions/All/1346-CheckIfNAndItsDoubleExist/CheckIfNAndItsDoubleExist.ts
export function checkIfExist(arr: number[]): boolean {
  const set = new Set<number>();

  for (let i = 0; i < arr.length; i++) {
    if (set.has(arr[i] * 2) || set.has(arr[i] / 2)) return true;
    else set.add(arr[i]);
  }

  return false;
}
